File: mobile/app/src/main/java/com/praeter/ui/mainactivity/fragment/classes/ClassAdapter.java
package com.praeter.ui.mainactivity.fragment.classes;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.praeter.R;
import com.praeter.data.local.model.Class;

import java.util.List;

public class ClassAdapter extends RecyclerView.Adapter<ClassViewHolder> {

    private final Context context;
    private final List<Class> classList;
    private final ClassClickListener listener;

    public ClassAdapter(@NonNull Context context,
                        @NonNull List<Class> classList,
                        @NonNull ClassClickListener listener) {
        this.context = context;
        this.classList = classList;
        this.listener = listener;
    }

    @Override
    public int getItemCount() {
        if (null != classList) {
            return classList.size();
        }
        return 0;
    }

    @NonNull
    @Override
    public ClassViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ClassViewHolder(context, LayoutInflater.from(parent.getContext())
                .inflate(R.layout.row_class_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ClassViewHolder holder, int position) {
        final Class item = classList.get(position);

        holder.bindData(item.getName(), item.getType());
        holder.itemCardView.setOnClickListener(
                view -> listener.onClassItemCLickListener(view, item, holder.getAdapterPosition()));
    }
}

File: mobile/app/src/main/java/com/praeter/ui/mainactivity/fragment/classes/ClassAdapter.kt
package com.praeter.ui.mainactivity.fragment.classes

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.praeter.R
import com.praeter.data.local.model.ClassModel
import com.praeter.databinding.RowAncientItemBinding
import com.praeter.databinding.RowClassItemBinding
import com.praeter.ui.mainactivity.fragment.metttheancient.AncientViewHolder

class ClassAdapter(
    private val context: Context,
    private val classModelList: List<ClassModel>,
    private val listener: ClassClickListener
) : RecyclerView.Adapter<ClassViewHolder>() {

    override fun getItemCount(): Int {
        return classModelList.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ClassViewHolder {
        val binding: RowClassItemBinding =
            RowClassItemBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )

        return ClassViewHolder(context, binding)
    }

    override fun onBindViewHolder(holder: ClassViewHolder, position: Int) {
        val item = classModelList[position]

        holder.bindData(item)
        holder.viewBinding.cvClassRowItem.setOnClickListener { view: View? ->
            listener.onClassItemCLickListener(
                view,
                item,
                holder.adapterPosition
            )
        }
    }
}